### ISSUE TITLE
the testing process too slow

### ISSUE THREAD

**Issue by dskuratovich** [qunitjs/qunit#1245]
hey everyone! 
I faced an issue regarding testing time. so, I have around 2.000 tests and they're so slow, I've found out that the problem is in the latency that was set in this method.

I've checked testing time without any latency and it takes around 25sec instead of 2-3 minutes. so, is there any chance to get some explanations why it was added? either add some configurable option in order to choose whether `begin()` will be run in async/sync mode.

here it is
https://github.com/qunitjs/qunit/blob/2999953e54ab5b1811d2f22e8c15997f00c42209/src/core.js#L271

well, perhaps I understand why it was added, just to avoid threads freezing in cases of long time running, but I hope there are some cases when we don't need such opportunity. 

@gibson042 so, what are your feelings on that?

**Comment by trentmwillis** [comment]
The delay in question is only 13ms and only occurs once when the test suite is started (either via `QUnit.start()` or `QUnit.load()`). It was added as a way to give asynchronously loaded tests time to be defined before starting the test suite.

This sounds to me like it may be something specific to your setup, as I am able to create [a simple test suite that runs 5000 tests in 1-2 seconds](https://jsbin.com/mukururama/1/edit?html,output). Maybe there is some code in the test setup that blocks the main thread preventing the `setTimeout` from being evaluated?

**Comment by rwjblue** [comment]
@trentmwillis - Seems like all usages  usage of `Test#resolvePromise` also has a 13ms delay:

https://github.com/qunitjs/qunit/blob/2999953e54ab5b1811d2f22e8c15997f00c42209/src/test.js#L778-L792

I believe that `assert.async()` also use that same infrastructure (`internalStop`), so it will be affected too.

**Comment by trentmwillis** [comment]
Yes, but that is a different block of code. Based on the initial comment, my understanding is that they removed the `setTimeout` from the start call and that dramatically reduced the test time.

**Comment by trentmwillis** [comment]
Note that I think we can likely speed up that timeout for the async/promise resolving case, though it sounds tangential to this issue.

**Comment by rwjblue** [comment]
Considering that many test harnesses leverage both a promise from `beforeEach` (2000 * 13ms) and another from the test itself (2000 * 13ms) then sprinkle in some `assert.async` usage in many tests (1200 * 13ms), and it isn’t terribly hard to get to around 60s spent waiting on what seems to be an arbitrary delay...

**Comment by rwjblue** [comment]
@trentmwillis - Ya, agreed. TBH, when I reviewed the original link on mobile I _thought_ it was the one in `internalStop`, which has given me pause recently so I piled on. 😜 happy to move to another issue if you’d prefer...

**Comment by rwjblue** [comment]
@trentmwillis - I've [updated your JSBin](https://jsbin.com/dahojib/edit?html,output) to `return Promise.resolve()`, and those same 5000 tests now take 104 seconds (up from 1 second)...

```js
    QUnit.module('5000 tests', function() {
      for (let i = 0; i < 5000; i++) {
        QUnit.test(`Test #${i}`, function(assert) {
          assert.ok(true);
          return Promise.resolve();
        });
      }
    });
```

**Comment by trentmwillis** [comment]
I tried doing some history searching to see if I could find a reason for the value `13` in both cases and can't seem to find anything. The closest I got was when `assert.async` was implemented and the comment is [actually to remove it](https://github.com/qunitjs/qunit/pull/653#discussion_r17195204). So I tried removing the `13` everywhere we use it and tests seem to still pass, so I've created a PR (#1246) to just remove them.

**Comment by platinumazure** [comment]
The number 13 is significant to me as the de facto minimum timeout in older versions of IE (and probably other browsers). But there's no reason other runtimes shouldn't be allowed to take advantage of quicker/more granular timers. (I have no idea if that is why 13 is used in the code, mind.)

**Comment by trentmwillis** [comment]
We've landed a commit to remove the delay parameter from the setTimeouts. This will go out in the next release and most folks should hopefully see a nice little speed bump :)

**Comment by dskuratovich** [comment]
Thanks so much guys! It's really cool

**Comment by dskuratovich** [comment]
by the way, @trentmwillis when we could expect a new release?

**Comment by trentmwillis** [comment]
I'm planning to do one sometime this week. Best way to get notified is to follow our [Twitter account](https://twitter.com/qunitjs).
